hoss/spatial.py

```diff
@@ -115,8 +116,22 @@ def get_spatial_index_ranges(
                         shape_file_path=shape_file_path,
                     )
                 )
-
-    return index_ranges
+        else:
+            override_dimensions = get_override_dimensions(varinfo, required_variables)
+            if len(override_dimensions) > 0:
+                for non_spatial_variable in non_spatial_variables:
+                    index_ranges.update(
+                        get_projected_x_y_index_ranges(
+                            non_spatial_variable,
+                            varinfo,
+                            dimensions_file,
+                            index_ranges,
+                            bounding_box=bounding_box,
+                            shape_file_path=shape_file_path,
+                            override_dimensions=override_dimensions,
+                        )
```

Comment:
I guess I could. Owen is suggesting to write a separate method and not re-use get_projected_x_y_index_ranges for the override case.. will check more into this

Reply:
You should definitely listen to @owenlittlejohns first, he understands this codebase far better than I do. My suggestions are more general and I think he has a better wholistic understanding of what should be done to improve this PR.

Reply:
This option is cleaner for this function itself, for sure, but my main concern is that `get_projected_x_y_index_ranges` is now a more complicated with branching, and forces things like writing dimensions to the prefetch NetCDF-4 file, likely could just be passed around in memory.

Reply:
I don't think I am writing dimensions to the prefetch NetCDF4 file.... We only get the coordinate variables in the prefetch file instead of the dimensions - since they don't exist

Reply:
https://github.com/nasa/harmony-opendap-subsetter/pull/15/commits/f836ee690d2433c52b2cb6fb8f0cac3b2d7f0e06

simplified the method

Reply:
Assuming this his handled elsewhere